lib/models/ObjectMDArchive.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * class representing the archive of object metadata.
+ *
+ * @class
+ */
+class ObjectMDArchive {
+    /**
+     *
+     * @constructor
+     * @param {Object} archiveInfo contains the archive info set by the TLP and returned by the TLP jobs
+     * @param {Date} [restoreRequestedAt] set at the time restore request is made by the client
+     * @param {Number} [restoreRequestedDays] set at the time restore request is made by the client
+     * @param {Date} [restoreCompletedAt] set at the time of successful restore
+     * @param {Date} [restoreWillExpireAt] computed and stored at the time of restore
+     * @throws {Error} case of invalid parameter
+     */
+    constructor(
+        archiveInfo,
+        restoreRequestedAt = undefined,
+        restoreRequestedDays = undefined,
+        restoreCompletedAt = undefined,
+        restoreWillExpireAt = undefined,
+    ) {
+        this.setArchiveInfo(archiveInfo);
+        this.setRestoreRequestedAt(restoreRequestedAt);
+        this.setRestoreRequestedDays(restoreRequestedDays);
+        this.setRestoreCompletedAt(restoreCompletedAt);
+        this.setRestoreWillExpireAt(restoreWillExpireAt);
+    }
+
+    /**
+     *
+     * @param {Object} data archiveInfo
+     * @returns {boolean} true if the provided object is valid
+     */
+    static isValid(data) {
+        try {
+            const object = new ObjectMDArchive(
+                data.archiveInfo,
+                data.restoreRequestedAt,
+                data.restoreRequestedDays,
+                data.restoreCompletedAt,
+                data.restoreWillExpireAt,
+            );
+            return !!object;
```

Comment:
Why not return true since the constructor will throw if it's invalid?
```suggestion
            return true;
```

Reply:
yes, done in https://github.com/scality/Arsenal/pull/1856/commits/de35c1e434dfe5f4e7eea348488e8433058ea8f8, along with some new tests to have a 100% code coverage on my additions: https://github.com/scality/Arsenal/pull/1856/commits/af71f12dbbf1918318809fc9aac2a8a25e8b5a3d